pyk/src/tests/integration/proof/test_imp.py

```diff
@@ -896,6 +896,36 @@ def test_all_path_reachability_prove(
             assert proof.status == proof_status
             assert leaf_number(proof) == expected_leaf_number
 
+    def test_terminal_node_subsumption(
```

Comment:
The test, in essence, runs a claim up to its terminal node, stops, then makes one more step, which only checks subsumption.